166 changing button labels to icons

amandahamynen opened the pull request:
# Changes
- Added bootstrap icons as global import in ```frontend/src/App.jsx```
- Changed ```Delete```-buttons into "trash can" -icons
- Changed ```Add ___```-buttons into "+"-buttons
- Added icons for elements in side navigation bar
- Added icons in top navigation bar for language selection, profile and logout, and flags for language options
- Changed "Flower page"-button into "info"-icon
- Changed "Edit"-button into "pencil"-icon
- Updated sorting arrows in flower lists
- Fixed unit tests that broke after adding icons

Closes #166

Eoyie reviewed (changes requested):
Great work!

The icons look great! I do have just a few comments though:

Unless we've been using clearly American English (I know in the code at least favourite is written as such, but I mean what is visible to the users), it would be nice to instead use the UK flag in the language menu.

Is there a reason why the modify buttons of "Save" and "Cancel" weren't changed at all. I also think generally, maybe outside of the flower list it would be nice if the buttons had both text and the icon. Perhaps ask the group opinion on this?


I think it is now a little unclear to the user that you can sort the tables. I tried looking at the available icons myself and found nothing that useful. If we had used bootstrap tables this might have not been an issue.. 

Either way I tried making it work with CSS. The CSS should be moved to a separate file, but here is what I quickly got working. 

    return (
      <span style={{ display: 'inline-block', position: 'relative', width: '1em', height: '1em' }}>
        <i className="bi bi-caret-down-fill" style={{ position: 'absolute', top: 7, fontSize: '0.8em' }}></i>
        <i className="bi bi-caret-up-fill" style={{ position: 'absolute', bottom: 0, fontSize: '0.8em' }}></i>
      </span>
    )

I accidentally also got the fields to stop moving on sorting with the width: '1em'. I think you could tweak some of the values as you see fit:

- Top: to move the lower icon down and create more space between the two (I don't think the upper icon should be moved?)
- fontSize: for the size of the icons, I don't think they look good being as big as the main sorting icons?

I also thought it looked better when there was at least one space between the field title and the icon, I simply added a space in the translations, but this does affect the texts in the flower modal as well. I'm not entirely sure where else this could be added, but I do think it looks a lot nicer. Perhaps simply padding or margin etc. could be used?

amandahamynen commented:
Thanks for the review!

> Unless we've been using clearly American English (I know in the code at least favourite is written as such, but I mean what is visible to the users), it would be nice to instead use the UK flag in the language menu.

I changed the flag from US -> UK like you requested!

> Is there a reason why the modify buttons of "Save" and "Cancel" weren't changed at all. I also think generally, maybe outside of the flower list it would be nice if the buttons had both text and the icon. Perhaps ask the group opinion on this?

I left "Save" and "Cancel" buttons as they were since I thought they look better without any icons, and updating the general look (colour etc) is part of the other UI task *Button improvements*? I decided to use only icons since I thought they looked better that way, but I'm fine with changing them to have both text and the icon if the rest of the group would prefer them that way. I saw that you already opened up a poll on discord, so now we could wait for everyone to give their input on this :)

Thanks for finding a solution for the sorting problem in the flowerlists! I applied the changes you suggested, and moved the styling into App.css. I tweaked some of the values a bit and added padding, let me know your thoughts on how they look now!
